lib/cb/responses/cover_letter/list.rb

```diff
@@ -0,0 +1,39 @@
+# Copyright 2015 CareerBuilder, LLC
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+module Cb
+  module Responses
+    module CoverLetter
+      class List < ApiResponse
+        private
+
+        def extract_models
+          [ cover_letters_node ].flatten.map { |api_cover_letter| Models::CoverLetter.new api_cover_letter }
+        end
+
+        def validate_api_hash
+          required_response_field(root_node, response)
+        end
+
+        def hash_containing_metadata
+          response
+        end
+
+        def cover_letters_node
+          response[root_node]['CoverLetters'].nil? ? [] : response[root_node]['CoverLetters']['CoverLetter']
```

Comment:
it's tested in the spec. when you have 0 cover letters the API returns null

On Thursday, September 24, 2015, Aaron Goldsmith <notifications@github.com>
wrote:

> In lib/cb/responses/cover_letter/list.rb
> <https://github.com/careerbuilder/ruby-cb-api/pull/215#discussion_r40355792>
> :
>
> > +        private
> > +
> > +        def extract_models
> > +          [ cover_letters_node ].flatten.map { |api_cover_letter| Models::CoverLetter.new api_cover_letter }
> > +        end
> > +
> > +        def validate_api_hash
> > +          required_response_field(root_node, response)
> > +        end
> > +
> > +        def hash_containing_metadata
> > +          response
> > +        end
> > +
> > +        def cover_letters_node
> > +          response[root_node]['CoverLetters'].nil? ? [] : response[root_node]['CoverLetters']['CoverLetter']
>
> Is there ever a case where the value at root_node could be nil outside of
> errors?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/careerbuilder/ruby-cb-api/pull/215/files#r40355792>.
>